File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        L = 0
        count = 0
        res = 0
        for R in range(len(nums)):
            if nums[R] == 0:
                count += 1
            
            while count > k:
                if nums[L] == 0:
                    count -= 1
                L += 1
            res = max(res, R - L + 1)
        return res

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int L = 0, res = 0;
        for (int R = 0; R < nums.size(); ++R) {
            int n = nums[R];
            k -= 1 - n;
            if (k < 0) {
                k += 1 - nums[L];
                L++;
            } else {
                res = max(res, R - L + 1);
            }
        }
        return res;
    }
};